nativelib/src/main/resources/scala-native/gc/commix/Phase.c

```diff
@@ -9,15 +9,34 @@
 #include <errno.h>
 #include <stdlib.h>
 
+/*
+If in OSX, sem_open cannot create a semaphore whose name is longer than
+PSEMNAMLEN characters. Else, we stick to POSIX compliance:
+
+  For POSIX.1-2017, XSI-conforming implementations shall set the
+  symbolic constant _XOPEN_UNIX to a value other than -1 and shall set the
+  symbolic constant _XOPEN_VERSION to the value 700.
+
+  The length of the name argument cannot exceed {_POSIX_PATH_MAX} on systems
+  that do not support the XSI option or exceed {_XOPEN_PATH_MAX} on XSI systems.
+*/
+#ifdef __APPLE__
+#include <sys/posix_sem.h>
+#define SEM_MAX_LENGTH PSEMNAMLEN
+#elif _XOPEN_UNIX != -1 && _XOPEN_VERSION == 700
+#define SEM_MAX_LENGTH _XOPEN_PATH_MAX
+else
+#define SEM_MAX_LENGTH _POSIX_PATH_MAX
+#endif
+
 void Phase_Init(Heap *heap, uint32_t initialBlockCount) {
     pid_t pid = getpid();
-    // size = static part + 20 digits (maximum 64 bit number) + null char
-    int workerNameSize = 11 + 20 + 1;
-    int masterNameSize = 11 + 20 + 1;
-    char startWorkersName[workerNameSize];
-    char startMasterName[masterNameSize];
-    snprintf(startWorkersName, workerNameSize, "commix_mt_%d", pid);
-    snprintf(startMasterName, masterNameSize, "commix_wk_%d", pid);
+    // size = platform dependent max semaphore length +
+    //        null character at the end
+    char startWorkersName[SEM_MAX_LENGTH + 1];
+    char startMasterName[SEM_MAX_LENGTH + 1];
+    snprintf(startWorkersName, SEM_MAX_LENGTH + 1, "mt_%d_commix", pid);
+    snprintf(startMasterName, SEM_MAX_LENGTH + 1, "wk_%d_commix", pid);
```

Comment:
I've moved the variable part to the beginning of the names, in case they need to be truncated at the end